x-pack/metricbeat/module/oracle/tablespace/temp_free_space.go

```diff
@@ -11,31 +11,24 @@ import (
 )
 
 type tempFreeSpace struct {
-	TablespaceName string
-	TablespaceSize sql.NullInt64
-	UsedSpaceBytes sql.NullInt64
-	FreeSpace      sql.NullInt64
-}
-
-func (d *tempFreeSpace) hash() string {
-	return d.TablespaceName
-}
-
-func (d *tempFreeSpace) eventKey() string {
-	return d.TablespaceName
+	TablespaceName  string
+	TotalSpaceBytes sql.NullInt64
+	UsedSpaceBytes  sql.NullInt64
+	FreeSpace       sql.NullInt64
 }
 
 func (e *tablespaceExtractor) tempFreeSpaceData(ctx context.Context) ([]tempFreeSpace, error) {
-	rows, err := e.db.QueryContext(ctx, "SELECT TABLESPACE_NAME, TABLESPACE_SIZE, ALLOCATED_SPACE, FREE_SPACE FROM DBA_TEMP_FREE_SPACE")
+	rows, err := e.db.QueryContext(ctx, `WITH sums AS ( SELECT (SELECT SUM(BYTES) FROM DBA_DATA_FILES) + (SELECT SUM(BYTES) FROM DBA_TEMP_FILES) AS TOTAL_SUM FROM dual ) SELECT t.TABLESPACE_NAME, s.TOTAL_SUM, t.ALLOCATED_SPACE, t.FREE_SPACE FROM DBA_TEMP_FREE_SPACE t, sums s `)
```

Comment:
Can you consider something simple as below


```
SELECT 
    t.TABLESPACE_NAME, 
    (SELECT SUM(BYTES) FROM DBA_DATA_FILES) + (SELECT SUM(BYTES) FROM DBA_TEMP_FILES) AS TOTAL_SUM, 
    t.ALLOCATED_SPACE, 
    t.FREE_SPACE 
FROM 
    DBA_TEMP_FREE_SPACE t;
```

Reply:
Talking about Niraj's query:

See [this](https://docs.oracle.com/cd/E17952_01/mysql-5.7-en/join.html#:~:text=INNER%20JOIN%20and%20%2C%20(comma)%20are%20semantically%20equivalent%20in%20the%20absence%20of%20a%20join%20condition%3A%20both%20produce%20a%20Cartesian%20product%20between%20the%20specified%20tables%20(that%20is%2C%20each%20and%20every%20row%20in%20the%20first%20table%20is%20joined%20to%20each%20and%20every%20row%20in%20the%20second%20table).).

With the cartesian product i.e., using commas makes this query incorrect? In this case, each row from DBA_TEMP_FREE_SPACE is combined with the single row from the sums subquery. In your query, it causes the TOTAL_SUM value to be repeated for each row of DBA_TEMP_FREE_SPACE, even though it should be a single value representing the total sum of bytes across all data files and temp files.

Try this query?

```
WITH sums AS (
  SELECT (SELECT SUM(BYTES) FROM DBA_DATA_FILES) + (SELECT SUM(BYTES) FROM DBA_TEMP_FILES) AS TOTAL_SUM
  FROM dual
)
SELECT t.TABLESPACE_NAME, (SELECT TOTAL_SUM FROM sums) AS TOTAL_SUM, t.ALLOCATED_SPACE, t.FREE_SPACE
FROM DBA_TEMP_FREE_SPACE t;
```

This query will calculate the total sum separately and display it as a single value for each row. No cartesian joins.